htdocs/map-leaflet.js

```diff
@@ -0,0 +1,525 @@
+            attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>',
+            subdomains: 'abcd',
+            maxZoom: 20
+        }
+    },
+    {
+        name: 'CartoDB Voyager',
+        url: 'https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png',
+        options: {
+            attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>',
+            subdomains: 'abcd',
+            maxZoom: 20
+        }
+    },
+    {
+        name: 'Stadia Alidade',
+        url: 'https://tiles.stadiamaps.com/tiles/alidade_smooth/{z}/{x}/{y}{r}.png',
+        config: {
+            maxZoom: 20,
+            noWrap: true,
+            attribution: '&copy; <a href="https://stadiamaps.com/">Stadia Maps</a>, &copy; <a href="https://openmaptiles.org/">OpenMapTiles</a> &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors',
+        },
+        info: 'In order to use Stadia maps, you must register. Once registered, you can whitelist your domain within your account settings.'
+    },
+    {
+        name: 'Stadia AlidadeDark',
+        url: 'https://tiles.stadiamaps.com/tiles/alidade_smooth_dark/{z}/{x}/{y}{r}.png',
+        config: {
+            maxZoom: 20,
+            noWrap: true,
+            attribution: '&copy; <a href="https://stadiamaps.com/">Stadia Maps</a>, &copy; <a href="https://openmaptiles.org/">OpenMapTiles</a> &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors'
+        },
+        info: 'In order to use Stadia maps, you must register. Once registered, you can whitelist your domain within your account settings.'
+    },
+];
+
+var mapExtraLayers = [
+    {
+        name: 'OpenWeatherMap',
+        url: 'https://tile.openweathermap.org/map/{layer}/{z}/{x}/{y}.png?appid={apikey}',
+        options: {
+            layer: 'clouds_new',
+            attribution: 'Map data: &copy; OpenWeatherMap'
+        },
+        depends: "weather_key"
+    },
+    {
+        name: 'OpenWeatherMap',
+        url: 'https://tile.openweathermap.org/map/{layer}/{z}/{x}/{y}.png?appid={apikey}',
+        options: {
+            layer: 'precipitation_new',
+            attribution: 'Map data: &copy; OpenWeatherMap'
+        },
+        depends: "weather_key"
+    },
+    {
+        name: 'WeatherRadar-USA',
+        url: 'http://mesonet.agron.iastate.edu/cache/tile.py/1.0.0/nexrad-n0q-900913/{z}/{x}/{y}.png',
+        options: {
+            attribution: 'Map data: &copy; Iowa State University'
+        },
+        depends: "!weather_key"
+    },
+    {
+        name: 'OpenSeaMap',
+        url: 'https://tiles.openseamap.org/seamark/{z}/{x}/{y}.png',
+        options: {
+            attribution: 'Map data: &copy; <a href="http://www.openseamap.org">OpenSeaMap</a> contributors'
+        },
+    },
+];
```

Comment:
I don't think these are really necessary. It's not that they hurt, but I'm pretty sure if you would like weather (or similar) information, you're better of to visit a separate service...

Reply:
removed the overlays for Leaflet in last commits.